src/codecTools/decodeCodec.ts

```diff
@@ -1,38 +1,46 @@
-// external
-import { fold } from 'fp-ts/lib/Either';
+import { fold, isLeft } from 'fp-ts/lib/Either';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as t from 'io-ts';
 
-// Determine the paths of the codec that are invalid
-const getPaths = <A>(v: t.Validation<A>): string[] =>
-  pipe(
+/**
+ * Determine the codec paths that are invalid.
+ * @param v - Validation errors.
+ * @returns The error strings formatted for human consumption.
+ */
+function getPaths<A>(v: t.Validation<A>): string[] {
+  return pipe(
     v,
     fold(
       (errors) =>
         errors.map((error) => {
-          const lastError = error.context[error.context.length - 1];
-          return `${error.context
-            .map((context) => context.key)
-            // Note: lastError.actual is helpful for debugging
-            .join('.')} expected type '${lastError.type.name}'`;
+          const lastCtx = error.context.at(-1);
+          const fullPath = error.context.map(({ key }) => key).join('.');
+          return `${fullPath} expected type '${lastCtx?.type.name}'`;
         }),
 
       () => ['no errors'],
     ),
   );
+}
 
-// Get custom error messages for the errors
-const getCustomErrors = <A>(
+/**
+ * Get custom error messages for the errors.
+ * @param v - Validation errors.
+ * @param customErrorFromContext - Custom mapper function.
+ * @returns The custom error strings formatted for human consumption.
+ */
+function getCustomErrors<A>(
   v: t.Validation<A>,
   customErrorFromContext: (validationContext: t.Context) => string,
-): string[] =>
-  pipe(
+): string[] {
+  return pipe(
```

Comment:
why the change from ` => ` to `{ return ...`? I'm surprised it didn't complain about this.

Reply:
So recently I've noticed that if you want docs for a function on hover, then you need them to be a full-fledged function -- considering the style earlier was creating an anonymous function, which ESLint does not enforce docs for, we were missing out on those docs, hence the change

Reply:
> Arrow functions and function functions are actually different in terms of how they handle references to this and arguments.

Yeah we don't typically use `this` references in our code so it was a safe change to make in this instance. Fair feedback though, I misspoke earlier -- I think we should have docs on our functions, and ESLint helps us enforce this if it knows they are functions